DataCapturing/Synchronization/Synchronizer.swift

```diff
@@ -149,6 +149,7 @@ public class Synchronizer {
                 return
             }
 
+            // TODO: This can lead to a crash as it is possible that two executions step over this.
```

Comment:
Since it does not belong with this branch (and we had only 2 crashes because of that within 18.000 Installations) I postponed a solution to a separate fix.